node-support/src/cloudevents.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function toCloudevent(metadata) {
+  return {
+    get specversion() {
+      return metadata["ce-specversion"];
+    },
+    get id() {
+      return metadata["ce-id"];
+    },
+    set id(id) {
+      metadata["ce-id"] = id;
+    },
+    get source() {
+      return metadata["ce-source"];
+    },
+    set source(source) {
+      metadata["ce-source"] = source;
+    },
+    get type() {
+      return metadata["ce-type"];
+    },
+    set type(type) {
+      metadata["ce-type"] = type;
+    },
+    get datacontenttype() {
+      return metadata["Content-Type"];
+    },
+    set datacontenttype(datacontenttype) {
+      metadata["Content-Type"] = datacontentype;
+    },
+    get dataschema() {
+      return metadata["ce-dataschema"];
+    },
+    set dataschema(dataschema) {
+      metadata["ce-dataschema"] = dataschema;
+    },
+    get subject() {
+      return metadata["ce-subject"];
+    },
+    set subject(subject) {
+      metadata["ce-subject"] = subject;
+    },
+    get time() {
+      return metadata["ce-time"];
+    },
+    set time(time) {
+      metadata["ce-time"] = time;
+    },
+    get entitykey() {
+      return metadata["ce-entitykey"];
+    },
```

Comment:
entity key actually isn't a Cloudevent property, I just introduced it here to carry our entity key. I had originally actually proposed an event key parameter for cloudevents:

https://github.com/cloudevents/spec/issues/209

This ended up being renamed to partitionkey, as the idea was that it would be used for partitioning in Kafka and similar partition supporting message brokers:

https://github.com/cloudevents/spec/pull/218

As such, while the partition key is almost always going to be the entity key, the partition key is not the right field to pass an entity key when what you need is the entity key, because they can differ.

All that said, sometime after I proposed the event key feature, a new field, `subject`, was added to cloud events:

https://github.com/cloudevents/spec/pull/406

Look at that, and also reading through the motivation and concrete examples in https://github.com/cloudevents/spec/issues/112, I think that field is an appropriate field to pt the entity key in. So I'm going to delete entity key and use subject instead.

Reply:
@jroper Yes, "subject" seems very fitting for our purposes here.